platform/mime/core/platform-mime-core-impl/src/main/java/ddf/mime/mapper/MimeTypeMapperImpl.java

```diff
@@ -175,6 +180,29 @@ public String guessMimeType(InputStream is, String fileExtension)
         // from being used when a CustomMimeTypeResolver may be more appropriate.
         List<MimeTypeResolver> sortedResolvers = sortResolvers(mimeTypeResolvers);
 
+        if (StringUtils.isEmpty(fileExtension)) {
+            try (TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream()) {
+                IOUtils.copy(is, tfbos);
+                try (InputStream inputStream = tfbos.asByteSource()
+                        .openStream()) {
+                    Parser parser = new AutoDetectParser();
+                    ToXMLContentHandler xmlContentHandler = new ToXMLContentHandler();
+                    Metadata metadata = new Metadata();
+                    parser.parse(inputStream, xmlContentHandler, metadata, new ParseContext());
```

Comment:
Though I don't know why I'm fully parsing it here instead of using the Tika detector that the parser uses. Changing it now